flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/TableEnvironment.scala

```diff
@@ -197,6 +209,17 @@ abstract class TableEnvironment(val config: TableConfig) {
     planner
   }
 
+  /**
+    * Returns the Calcite [[org.apache.calcite.rel.`type`.RelDataTypeFactory]]
+    * of this TableEnvironment. */
+  private[flink] def getTypeFactory: RelDataTypeFactory = {
```

Comment:
`getTypeFactory()` and `getTable()` are only called to create a `CatalogNode` which internally obtains the `rowType: RelDataType` of the table. I think we can simply return the `rowType` for a given table name and remove `getTypeFactory()` and `getTable()` from this class.